src/hazelcore/semantics/CursorInfo.re

```diff
@@ -146,7 +146,7 @@ module Pat = {
          )
```

Comment:
I started looking in to this and I found that the code [here](https://github.com/hazelgrove/hazel/blob/fix-tuple-type-ambiguity/src/hazelcore/semantics/CursorInfo.re#L847) assumes `ty` is a tuple type, which is not the case (this does not cause an error, because `HTyp.get_prod_elements` (and similar for `UHTyp`, `UHPat` etc.) just creates a singleton list of the given type when the given type is not a product type. It feels like a nasty shortcut and has been a bit of a thorn in my eye since starting this refactoring, but sadly the function is quite widespread and I don't feel like I'm up to the task of removing it everywhere) (sorry for the rant :sweat_smile:).
Anyway, I started searching for how errors are done for arbitrary operators, but I found `ana_cursor_info_zopseq` really only considers tuple `OpSeq`s and when the operator is not `Comma` it just switches to synthetic cursor info, which just says `got as expected` :(. I'm not sure why it does this.

I think this case should generate some kind of `AnaTypeInconsistent` info, but I wanted to pass this by you before I started working on it.

Reply:
once we add labeled tuples, we're definitely going to want to distinguish 1-tuples from non-tuples. this is probably the right time to fix that. can you take a shot at fixing this @cdfa? @dmoon1221 can you assist as necessary?

Reply:
yes this case should generate AnaTypeInconsistent. the code you linked to above is the right place to fix it.

Reply:
I'm not sure if this is the right time to fix it. This PR is already quite big and it's generating more and more merge conflicts. To be frank, I also just don't have the time to do a full refactor for this.
What I could do is make the functions fail explicitly when its argument is not a tuple, so at least places like this where the assumption of a tuple type is incorrect will become more obvious.

Reply:
This specific problem of 1-tuples is fixed in acb63046b6c1c02ddfd7ce995ba810b1be3723e3